include/glow/Base/Tensor.h

```diff
@@ -290,14 +301,16 @@ class Tensor final {
     unownedTensor.unpaddedSize_ = unpaddedSize_;
 
     if (offsets.size() == 0) {
-      assert(size() == unownedTensor.size() && "The size of the unowned tensor "
-                                               "should the same as the size of "
-                                               "the original tensor");
+      assert(actualSize() == unownedTensor.actualSize() &&
```

Comment:
why is size() not sufficient here?

Reply:
Unowned tensor assumes that no data was copied. If actual size is different, then strides were changed, then probably we may address "garbage" element somewhere, which is unwanted.

In any way, how could this happen? When we create unowned tensor, we don't even pass new strides. It's just safer to verify more (that actual sizes match).

Reply:
Wait so because we are operating with offsets from `firstElemPtr` here, getUnowned won't take strides into account and we'll end up with garbage values

Reply:
Good catch. Fixed now.